### ISSUE TITLE
Soft Delete Restore Action button on pageview

### ISSUE THREAD

**Issue by MarJose123** [filamentphp/filament#3364]
### Package

filament/filament

### Package Version

v2

### Laravel Version

v9

### Livewire Version

v2

### PHP Version

v8

### Problem description

Filament\Pages\Actions\RestoreAction::Filament\Pages\Actions\{closure}(): Argument #1 ($record) must be of type Illuminate\Database\Eloquent\Model, null given

![image](https://user-images.githubusercontent.com/18107626/182577491-d5fc76bc-6a76-410f-a310-2f940ce03c06.png)

### Expected behavior

it should prompt a notification restored something.

### Steps to reproduce

make sure your page has enabled the soft-delete.
1. add the `Actions\RestoreAction::make()->icon('heroicon-o-trash')` on your ViewPage.
2. Create record
3. Delete the record
4. restore the deleted record

### Reproduction repository
https://github.com/whoami213/FilamentPHP-ViewPage-Error

### Relevant log output

```shell
Filament\Pages\Actions\RestoreAction::Filament\Pages\Actions\{closure}(): Argument #1 ($record) must be of type Illuminate\Database\Eloquent\Model, null given, called in /home/unhex/self_project/eBMS-v3/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php on line 36
```

**Comment by whoami213** [comment]
Hi @danharrin can you check if this something a totally issues on my end only?

**Comment by danharrin** [comment]
Yeah, we can have a look if you submit the reproduction repository URL, just edit your original message

**Comment by whoami213** [comment]
I've update the repo link

**Comment by ralphjsmit** [comment]
I also experience this issue with the DeleteAction. It's quite serious, since it breaks the Delete buttons in entire applications.

@danharrin I can share a Flare stacktrace if you want.

**Comment by danharrin** [comment]
Fixed by #3587.
